### PR TITLE
SQ-63811 Ajustar components do NGX-CSS para funcionar em SSR

### PR DESCRIPTION
**Pull request by CaioDalo**
[SQ-63811](https://duopana.myjetbrains.com/youtrack/issue/SQ-63811/Ajustar-components-do-NGX-CSS-para-funcionar-em-SSR)

### CODE REVIEW

**Comment by CaioDalo** [comment]
@coderabbitai review

File: application/src/app/app.component.ts [hunk 1]
```diff
@@ -7,18 +9,24 @@ import { Component, OnInit } from '@angular/core'
...
+ngOnInit() {
+    this.theme = !this.isServer && localStorage.getItem('theme') || this.theme
+    const html = this.document.getElementsByTagName('html')[0]
     html.classList.value = `${this.theme}`
   }
 
   toggleTheme() {
     this.theme = this.theme === 'dark' ? 'light' : 'dark'
-    const html = document.getElementsByTagName('html')[0]
+    const html = this.document.getElementsByTagName('html')[0]
```
[the remark this answers is not in the document]
Comment (CaioDalo) [reply application/src/app/app.component.ts:29]: This application folder is just a mock to test the components and ensure that the components is working well in ssr

File: application/src/app/components/header/header.component.ts [hunk 2]
```diff
@@ -7,11 +8,15 @@ import { Component, OnInit } from '@angular/core'
...
+
+  constructor(@Inject(PLATFORM_ID) private platformId: InjectionToken<Object>, @Inject(DOCUMENT) private _doc: Document) {}
+
+  theme = !this.isServer && window.matchMedia('(prefers-color-scheme:dark)').matches ? 'dark' : 'light'
 
   ngOnInit() {
-    this.theme = localStorage.getItem('theme') || this.theme
-    const html = document.getElementsByTagName('html')[0]
+    this.theme = !this.isServer && localStorage.getItem('theme') || this.theme
+    const html = this._doc.getElementsByTagName('html')[0]
```
[the remark this answers is not in the document]
Comment (CaioDalo) [reply application/src/app/components/header/header.component.ts:19]: This application folder is just a mock to test the components and ensure that the components is working well in ssr

**Comment by CaioDalo** [comment]
@coderabbitai review

**Review by JonasPeres — DISMISSED** [review]

**Comment by CaioDalo** [comment]
@vinimarcili code review

File: application/src/app/app.component.html [hunk 3]
```diff
@@ -2,6 +2,6 @@
   <main-header></main-header>
   <main>
     <router-outlet></router-outlet>
-    <!-- Use Here to test library components -->
+    <sq-input></sq-input>
```
Comment (vinimarcili) [inline application/src/app/app.component.html:5]: volta o anterior aqui

File: application/src/app/app.component.ts [hunk 4]
```diff
@@ -7,19 +9,27 @@ import { Component, OnInit } from '@angular/core'
...
 
+  constructor(@Inject(PLATFORM_ID) private platformId: InjectionToken<Object>,
+  @Inject(DOCUMENT) private document: Document,
+  public getWindow: GetWindow
+) {}
 
-  ngOnInit() {
-    this.theme = localStorage.getItem('theme') || this.theme
-    const html = document.getElementsByTagName('html')[0]
+theme = !this.isServer && window.matchMedia('(prefers-color-scheme:dark)').matches ? 'dark' : 'light'
```
Comment (vinimarcili) [inline application/src/app/app.component.ts:19]: pq precisa do isServer? vc já não ta injetando o window?

File: application/src/app/components/header/header.component.ts [hunk 5]
```diff
@@ -7,18 +8,24 @@ import { Component, OnInit } from '@angular/core'
...
+  isServer = isPlatformServer(this.platformId)
+
+  constructor(@Inject(PLATFORM_ID) private platformId: InjectionToken<Object>, @Inject(DOCUMENT) private _doc: Document) {}
+
+  theme = !this.isServer && window.matchMedia('(prefers-color-scheme:dark)').matches ? 'dark' : 'light'
 
   ngOnInit() {
-    this.theme = localStorage.getItem('theme') || this.theme
-    const html = document.getElementsByTagName('html')[0]
+    this.theme = !this.isServer && localStorage.getItem('theme') || this.theme
```
Comment (vinimarcili) [inline application/src/app/components/header/header.component.ts:18]: melhor que botar o isServer é melhor fazer um localStorage?.getItem

**Review by vinimarcili — CHANGES_REQUESTED** [review]

**Review by JonasPeres — DISMISSED** [review]

**Review by vinimarcili — APPROVED** [review]

**Review by vinimarcili — APPROVED** [review]

**Pull request merged**